src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -120,64 +125,121 @@ def update_env(spec, env):
                 container["env"] = env
 
 
-def update_resources(spec, min_cpu, max_cpu, min_memory, max_memory, gpu):
+def update_resources(
+    spec,
+    min_cpus,
+    max_cpus,
+    min_memory,
+    max_memory,
+    custom_resources,
+):
     container = spec.get("containers")
     for resource in container:
         requests = resource.get("resources").get("requests")
         if requests is not None:
-            requests["cpu"] = min_cpu
+            requests["cpu"] = min_cpus
             requests["memory"] = min_memory
-            requests["nvidia.com/gpu"] = gpu
+            for k in custom_resources.keys():
+                requests[k] = custom_resources[k]
         limits = resource.get("resources").get("limits")
         if limits is not None:
-            limits["cpu"] = max_cpu
+            limits["cpu"] = max_cpus
             limits["memory"] = max_memory
-            limits["nvidia.com/gpu"] = gpu
+            for k in custom_resources.keys():
+                limits[k] = custom_resources[k]
+
+
+def head_worker_gpu_count_from_cluster(
+    cluster: "codeflare_sdk.cluster.Cluster",
+) -> typing.Tuple[int, int]:
+    head_gpus = 0
+    worker_gpus = 0
+    for k in cluster.config.head_custom_resource_requests.keys():
+        resource_type = cluster.config.custom_resource_mapping[k]
+        if resource_type == "GPU":
+            head_gpus += cluster.config.head_custom_resource_requests[k]
```

Comment:
This function fails when running `get_cluster()` with accelerators specified with the below error.
```
 unsupported operand type(s) for +=: 'int' and 'str'
```
We could force the type conversion here